src/swarm/node/model/NeoNode.d

```diff
@@ -845,13 +891,18 @@ public class NodeBase ( ConnHandler : ISwarmConnectionHandler ) : INodeBase
         }
 
         // Super ctor.
+        InetAddress!(false) addr, neo_addr;
+        addr = cast(sockaddr_in)node;
+        neo_addr = addr;
+        neo_addr.port = neo_port;
+
         super(node, conn_setup_params,
             this.listener = new Listener(
-                addr(node.Address, node.Port), this.socket, conn_setup_params,
+                cast(sockaddr*)&addr.addr, this.socket, conn_setup_params,
```

Comment:
This is fine, considering that `addr.addr` is `sockaddr_in`. However, I think by casting pointer we're loosing the type safety here. I would just add the static assert before `static assert(typeof(addr.addr) is sockaddr_in)`.